connector/src/main/scala/com/google/cloud/spark/bigquery/SparkBigQueryOptions.scala

```diff
@@ -270,6 +242,66 @@ object SparkBigQueryOptions {
     getAnyOption(globalOptions, options, name)
       .map(_.toBoolean)
       .getOrElse(defaultValue)
-
 }
 
+  case class IntermediateFormat(dataSource: String,
+                                formatSuffix: String,
+                                formatOptions: FormatOptions)
+  object IntermediateFormat {
+    private val PermittedIntermediateFormats = Map("avro" -> FormatOptions.avro(),
+      "parquet" -> FormatOptions.parquet(),
+      "json" -> FormatOptions.json(),
+      "orc" -> FormatOptions.orc())
+
+    def apply (format: String, sparkVersion: String, sqlConf: SQLConf) : IntermediateFormat = {
+      if (!PermittedIntermediateFormats.contains(format)) {
+        throw new IllegalArgumentException(
+          s"""Data read format '${format}' is not supported.
+             |Supported formats are '${PermittedIntermediateFormats.keys.mkString(",")}'"""
+            .stripMargin.replace('\n', ' '))
+      }
+
+      val formatSuffix = if (format == "json") {
+        ".json"
+      } else {
+        s".${PermittedIntermediateFormats(format).getType.toLowerCase}"
+      }
+
+      if (format == "avro") {
+        val dataSource = if (sparkVersion >= "2.4") {
+          format
+        } else {
+          "com.databricks.spark.avro"
+        }
+
+        try {
+          DataSource.lookupDataSource(dataSource, sqlConf)
+        } catch {
+          case re: org.apache.spark.sql.AnalysisException =>
```

Comment:
Add import for `AnalysisException`, rename to ae (re was for `RuntimeException`)